### ISSUE TITLE
AWS: out of instances

### ISSUE THREAD

**Issue by janscholten** [HPCE/hpce-2017-cw4#32]
Related to #24,
I have tried now for a couple of times to fire up an c4.8xlarge, however, Amazon is out of instances:

> Launch Failed
> We currently do not have sufficient c4.8xlarge capacity in zones with support for 'gp2' volumes. Our system will be working on provisioning additional capacity.

What should I do? I don't want to hang around all day until they have capacity, nor do I want to bias my results by selecting a different instance.

**Comment by janscholten** [comment]
I've tried again using a different region (Canada instead of N. Cali) but that didn't solve or change the problem.

**Comment by janscholten** [comment]
Now I chose London and that works!

**Comment by janscholten** [comment]
Now I chose London and that works!

**Comment by m8pple** [comment]
Interesting - I chose that instance as it is _usually_ fairly uncontested, so there haven't
been any particular provisioning problems in the past. The spot price history in
North Virginia also looks reasonable:
![image](https://user-images.githubusercontent.com/3024705/32556650-e56b3d36-c497-11e7-8f5c-e165ec89595c.png)

**Comment by m8pple** [comment]
(Reopening as it is kind of interesting)

**Comment by m8pple** [comment]
Ah, just noticed that you were talking about the North California region. Looking at _their_
history we see something quite different:
![image](https://user-images.githubusercontent.com/3024705/32556861-76b56550-c498-11e7-95e8-53115e3e01da.png)

I think what we're seeing is supply and demand in action - people are bidding _above_ the
on-demand price on the spot market, so it makes sense for Amazon to give instances to
the spot market, where they earn more money.

The advantage of the on-demand instances is that anyone who already has an instance
running won't have their instance terminated. But that graph suggests that a whole
bunch of people with spot instances probably had their instances terminated because
they didn't bid enough.

**Comment by m8pple** [comment]
Also, on the economics side of things still, note that prices actually vary
by region (try switching the region):
https://aws.amazon.com/ec2/pricing/on-demand/

North California is the most expensive US region (I think Tokyo is more
expensive again). Which leads to two interesting questions:
- Why are some locations more expensive than others for AWS (think property
   prices, energy prices, ...)?
- Why are people will to pay _so much_ to have their instances in the
  North California region? Or, what is potentially [so great about London for you](http://www.cloudping.info/)?

**Comment by janscholten** [comment]
Ah, thank you for clarifying! It's quite interesting indeed, and it also explains why people are overbidding the on-demand price right? Furthermore, I wasn't aware paying 10x more for california than e.g. Ohio, so i'll keep that in mind.

**Comment by m8pple** [comment]
Yeah, I think I'd always looked at things from the users point of view before. So when
I saw people bidding spot prices over the on-demand price I always thought it was really
odd, as I thought "why don't they just go to on-demand"? It had never actually occurred
to me that AWS can balance their fixed pool of machines between spot and on-demand,
so this behaviour is completely expected.

So previously I was committing the fallacy of considering the users as rational
actors, but not considering Amazon as a rational actor. I've discussed the odd spot
price behaviour with people before, and we just never seemed to twig that they
could just deny _new_ on-demand users in order to exploit the excess spot, without
affecting existing on-demand users.

We're all still learning :)
